src/SonarScanner.MSBuild.Tasks/IsTestFileByName.cs

```diff
@@ -68,52 +66,29 @@ public sealed class IsTestFileByName : Task
 
         public override bool Execute()
         {
-            var taskSuccess = true;
-
-            var config = TaskUtilities.TryGetConfig(AnalysisConfigDir, new MSBuildLoggerAdapter(Log));
-
-            if (config != null)
+            if (TaskUtilities.TryGetConfig(AnalysisConfigDir, new MSBuildLoggerAdapter(Log)) is AnalysisConfig config)
             {
-                var regEx = TryGetRegularExpression(config);
-
-                try
+                if (config.GetAnalysisSettings(true).TryGetValue(TestRegExSettingId, out var regEx) && !string.IsNullOrWhiteSpace(regEx))
                 {
-                    // Let's use a case sensitive regex (default behavior)
-                    IsTest = Regex.IsMatch(FullFilePath, regEx);
+                    try
+                    {
+                        // Let's use a case sensitive regex (default behavior)
+                        IsTest = Regex.IsMatch(FullFilePath, regEx);
+                    }
+                    catch (ArgumentException ex) // thrown for invalid regular expressions
+                    {
+                        Log.LogError(Resources.IsTest_InvalidRegularExpression, regEx, ex.Message, TestRegExSettingId);
+                    }
                 }
-                catch (ArgumentException ex) // thrown for invalid regular expressions
+                else
                 {
-                    taskSuccess = false;
-                    Log.LogError(Resources.IsTest_InvalidRegularExpression, regEx, ex.Message, TestRegExSettingId);
+                    Log.LogMessage(MessageImportance.Low, Resources.IsTest_NameNotChecked);
```

Comment:
should this be low or normal? (I'm thinking that users that might be used to the old test detection would be helped by a "normal" log - if they actually look at the logs, that is)

Reply:
The `NameNotChecked` is the normal state now so it would be unneeded noise.